test/src/unit/modules/Project.test.js

```diff
@@ -0,0 +1,150 @@
+            const project = new Project({
+                name: 'newdummyproject'
+            }, './someworkspace');
+            project.should.be.an('object');
+            const id = project.projectID;
+            const name = project.name;
+            project.should.have.property('directory');
+            project.directory.should.equal(`${name}-${id}`);
+        });
+        it('Correctly initialises using a comprehensive list of arguments', () => {
+            const args = {
+                projectID: '9318ab10-fef9-11e9-8761-9bf62d92b58b',
+                name: 'my-java-prject',
+                codewindVersion: 'someversion',
+                language: 'java',
+                locOnDisk: 'location',
+                workspace: '/codewind-workspace/',
+                directory: 'directorythatisntname-id',
+                infLockFlag: false,
+                startMode: 'run',
+                projectType: 'docker',
+                buildLogPath: '/codewind-workspace/.logs/my-java-prject-9318ab10-fef9-11e9-8761-9bf62d92b58b-9318ab10-fef9-11e9-8761-9bf62d92b58b/docker.build.log',
+                state: 'open',
+                autoBuild: false
+                
+            }
+            const project = new Project(args, '/codewind-workspace/');
+            project.should.be.an('object');
+            project.should.containSubset(args);
+            const id = project.projectID;
+            const name = project.name;
+            project.directory.should.not.equal(`${name}-${id}`);
+        });
+        it('Correctly sets directory to name-id', () => {
+            const project = new Project({
+                name: 'newdummyproject'
+            }, './someworkspace');
+            project.should.be.an('object');
+            const id = project.projectID;
+            const name = project.name;
+            project.should.have.property('directory');
+            project.directory.should.equal(`${name}-${id}`);
+        });
+    })
+    describe('toJSON', () => {
+        it('Checks that toJSON removes fields which shouldn\'t be written to .info file on disk', () => {
+            const obj = {
+                logStreams: 'logstream',
+                loadInProgress: true,
+                loadConfig: 'someconfig'
+            }
+            const project = new Project({ name: 'dummy'}, '/codewind-workspace/');
+            project.should.be.an('object');
+            project.should.containSubset({ name: 'dummy' });
+            project.should.not.containSubset(obj);
+    
+            project.logStreams = obj.logStreams,
+            project.loadInProgress = obj.loadInProgress;
+            project.loadConfig = obj.loadConfig;
+    
+            project.should.containSubset(obj);
+    
+            const json = project.toJSON();
+            json.should.not.containSubset(obj);
+        });
+    })
+    describe('getPort', () => {
+        it('Checks that the port returned is internal as RUNNING_IN_K8S is false', () => {
```

Comment:
You could separate these into 2 describe blocks. i.e 1 for `RUNNING_IN_K8S` is false, and the other for the reverse. I'd say that'd make the logs a bit easier to follow, and mean adding new tests simpler